modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,188 @@
+    $field_elements = $processed_object['#fieldgroups'][$element['#group_name']]->children;
+    $links = [];
+    foreach ($field_elements as $value) {
+      // Check if we should add the field to page contents.
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
+        continue;
+      }
+      // We will add anchors when the element, field or group, has a label.
+      $label = $this->fieldGetLabel($value, $processed_object);
+      if ($label === '') {
+        continue;
+      }
+      $type = $this->groupOrField($value, $processed_object['#fieldgroups']);
+      // If a group has a default id, use it as anchor link.
+      if ($type === 'group' && !empty($processed_object['#fieldgroups'][$value]->format_settings['id'])) {
+        $id = $processed_object['#fieldgroups'][$value]->format_settings['id'];
+      }
+      else {
+        $id = "inline-nav-" . $i = isset($i) ? ++$i : 1;
+      }
+
+      $links[] = [
+        'href' => "#" . $id,
+        'label' => $label,
+      ];
+      $element[$value]['#id'] = $id;
+      $element[$value]['#attributes']['id'] = $id;
+    }
+
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if we should exclude this field from the in-page navigation menu.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $field_groups
+   *   List of groups.
+   *
+   * @return bool
+   *   Exclude or not the field.
+   */
+  protected function excludeField(string $field_name, array $field, array $field_groups) :bool {
+    $type = $this->groupOrField($field_name, $field_groups);
+    switch ($type) {
+      case "field":
+        // Exclude empty fields.
+        if (!isset($field['#items'])) {
+          return TRUE;
+        }
+
+        break;
+
+      case 'group':
+        // Exclude empty groups.
+        if (count($field) < 1) {
+          return TRUE;
+        }
+
+        foreach ($field as $value) {
+          // If there is a field with values, don't exclude the group.
+          if (isset($value['#items'])) {
+            return FALSE;
+          }
+        }
+
+        // Exclude groups with only empty fields.
+        return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Get the label used on the field.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $processed_object
+   *   Entity processed.
+   *
+   * @return string
+   *   Label of the field.
+   */
+  protected function fieldGetLabel(string $field_name, array $processed_object): string {
```

Comment:
Remove this method, as we will have one dedicated to establishing if the item should be included in the navigation or not (see above).

Reply:
Removed